opensrp-app/src/main/java/org/smartregister/repository/AllSharedPreferences.java

```diff
@@ -51,11 +53,16 @@ public void updateANMUserName(String userName) {
     public String fetchRegisteredANM() {
         return preferences.getString(ANM_IDENTIFIER_PREFERENCE_KEY, "").trim();
     }
+    public void updateANMRole(String role) {
+        preferences.edit().putString(ANM_IDENTIFIER_ROLE, role).commit();
+    }
 
     public boolean isRegisteredANM(String userName) {
         return preferences.getStringSet(ANM_IDENTIFIER_SET_PREFERENCE_KEY, new HashSet<>()).contains(userName);
     }
-
+    public String fetchRegisteredRole() {
+        return preferences.getString(ANM_IDENTIFIER_ROLE, "").trim();
+    }
```

Comment:
What is the use case for this?

Reply:
From App side it's return whatever role is saved at userService saveUserInfo method
` public void saveUserInfo(User user) {
        try {
            if (user != null && user.getPreferredName() != null) {
                String preferredName = user.getPreferredName();
                String userName = user.getUsername();
                allSharedPreferences.updateANMPreferredName(userName, preferredName);
                if(user.getRoles() !=null && user.getRoles().size() >0){
                    String defRole = user.getRoles().get(0);
                    allSharedPreferences.updateANMRole(defRole);
                }
            }
        } catch (Exception e) {
            Timber.e(e);
        }

        String userInfoString = AssetHandler.javaToJsonString(user);
        saveUserInfoTask.save(userInfoString);
    }`
line 571 to 573